vocabularies/DirectEdit.xml

```diff
@@ -0,0 +1,77 @@
+  </edmx:Reference>
+  <edmx:Reference Uri="https://oasis-tcs.github.io/odata-vocabularies/vocabularies/Org.OData.Validation.V1.xml">
+    <edmx:Include Alias="Validation" Namespace="Org.OData.Validation.V1" />
+  </edmx:Reference>
+  <edmx:Reference Uri="https://sap.github.io/odata-vocabularies/vocabularies/Common.xml">
+    <edmx:Include Namespace="com.sap.vocabularies.Common.v1" Alias="Common" />
+  </edmx:Reference>
+
+  <edmx:DataServices>
+    <Schema xmlns="http://docs.oasis-open.org/odata/ns/edm" Namespace="com.sap.vocabularies.DirectEdit.v1" Alias="DirectEdit">
+      <Annotation Term="Core.Description">
+        <String>Terms for Direct-Edit User Interfaces</String>
+      </Annotation>
+      <Annotation Term="Core.Description" Qualifier="Published">
+        <String>2020-07-10 © Copyright 2020 SAP SE. All rights reserved</String>
+      </Annotation>
+      <Annotation Term="Core.Links">
+        <Collection>
+          <Record>
+            <PropertyValue Property="rel" String="latest-version" />
+            <PropertyValue Property="href" String="https://sap.github.io/odata-vocabularies/vocabularies/DirectEdit.xml" />
+          </Record>
+          <Record>
+            <PropertyValue Property="rel" String="alternate" />
+            <PropertyValue Property="href" String="https://sap.github.io/odata-vocabularies/vocabularies/DirectEdit.json" />
+          </Record>
+          <Record>
+            <PropertyValue Property="rel" String="describedby" />
+            <PropertyValue Property="href" String="https://github.com/sap/odata-vocabularies/blob/master/vocabularies/DirectEdit.md" />
+          </Record>
+        </Collection>
+      </Annotation>
+
+      <Term Name="SideEffects" Type="DirectEdit.SideEffectsType" Nullable="false" AppliesTo="EntitySet">
+        <Annotation Term="Common.Experimental" />
+        <Annotation Term="Core.Description" String="Determine side effects of client-side data modification" />
+      </Term>
+
+      <ComplexType Name="SideEffectsType">
+        <Annotation Term="Common.Experimental" />
+        <Annotation Term="Core.LongDescription">
+          <String>A direct-edit side-effect consists of:
+
+- a list of possible triggers, _preliminary_ syntax:
+  - `Path:` followed by a path to the property whose change triggers the side-effects determination 
+  - `Action:` followed by the name of the action that is called for the side-effects determination
+
+- an action bound to the entity set with the following non-binding parameters:
+  - `Qualifier` of type [`Core.SimpleIdentifier`](https://github.com/oasis-tcs/odata-vocabularies/blob/master/vocabularies/Org.OData.Core.V1.md#SimpleIdentifier)
+    or cast-compatible: the qualifier of the `SideEffects` annotation
+  - `Trigger` of type `Edm.String`: the trigger of side-effects determination, see `Triggers` property
+  - `Data` of either the entity type of the annotated entity set or a complex type that is structure-compatible with it, 
```

Comment:
Is it sufficient to pass only one entity at a time? Can't changes to _other_ entities which the client has accumulated also affect the outcome of the determination? In other words: Must `Data` not have the structure of the entire business object?

Reply:
We allow passing trees. Either directly as nested data of the entity type or via a complex type with nested complex types.
We do not currently have the use case of multiple top level entities although this could in theory happen. In that case we could still extend the action signature.